bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -23,81 +26,95 @@
 #define WINDOWS8_OR_GREATER "windows8_or_greater"
 
 #define BOOTLOADER_SERVER_PORT 5001
-#define ISLAND_SERVER_PORT ":5000"
 
-void error(const char *msg) { perror(msg); exit(1); }
+int shouldMonkeyRun(char* windowsVersion) {
+    if (strcmp(windowsVersion, XP_OR_LOWER))
+        return 1;
+    if (strcmp(windowsVersion, VISTA))
+        return 1;
+    if (strcmp(windowsVersion, VISTASP1))
+        return 1;
+    if (strcmp(windowsVersion, VISTASP2))
+        return 1;
+    else
+        return 0;
+}
 
-char** getIpAddresses(int *addrCount, char** hostname){
+char** getIpAddresses(int *addrCount, char** hostname) {
     int j = 0;
+    *addrCount = j;
+    *hostname = (char *)malloc(2);
+    if (*hostname == NULL) {
+        error("Malloc failure!");
+    }
+    strcpy_s(*hostname, 1, "");
 
     DWORD Err;
     PFIXED_INFO pFixedInfo;
-    DWORD FixedInfoSize = 0;
+    ULONG FixedInfoSize = 0;
 
     PIP_ADAPTER_INFO pAdapterInfo, pAdapt;
     DWORD AdapterInfoSize;
     PIP_ADDR_STRING pAddrStr;
 
-    //
-    // Get the main IP configuration information for this machine using a FIXED_INFO structure
-    //
-	Err = GetNetworkParams(NULL, &FixedInfoSize);
-    if (Err != 0)
-    {
-        if (Err != ERROR_BUFFER_OVERFLOW)
-            error("GetNetworkParams sizing failed\n");
-    }
-
-    // Allocate memory from sizing information
-	pFixedInfo = (PFIXED_INFO)malloc(FixedInfoSize);
-    if (pFixedInfo == NULL)
+    pFixedInfo = (FIXED_INFO *) malloc(sizeof(FIXED_INFO));
+    if (pFixedInfo == NULL) {
         error("Memory allocation error\n");
+    }
+    FixedInfoSize = sizeof (FIXED_INFO);
+
+    if (GetNetworkParams(pFixedInfo, &FixedInfoSize) == ERROR_BUFFER_OVERFLOW) {
+        free(pFixedInfo);
+        pFixedInfo = (FIXED_INFO *) malloc(FixedInfoSize);
+        if (pFixedInfo == NULL) {
+            printf("Error allocating memory needed to call GetNetworkParams\n");
```

Comment:
Some allocation errors you call `error`, sometimes you return a value. Be consistent.